src/main/java/org/opentripplanner/apis/gtfs/mapping/RouteRequestMapper.java

```diff
@@ -40,227 +43,51 @@ public static RouteRequest toRouteRequest(
-          )
-      );
-      preferences.withTransit(tr -> {
-        callWith.argument("boardSlack", tr::withDefaultBoardSlackSec);
-        callWith.argument("alightSlack", tr::withDefaultAlightSlackSec);
-        callWith.argument(
-          "preferred.otherThanPreferredRoutesPenalty",
-          tr::setOtherThanPreferredRoutesPenalty
-        );
-        // This is deprecated, if both are set, the proper one will override this
-        callWith.argument(
-          "unpreferred.useUnpreferredRoutesPenalty",
-          (Integer v) -> tr.setUnpreferredCost(CostLinearFunction.of(Duration.ofSeconds(v), 0.0))
-        );
-        callWith.argument("unpreferred.unpreferredCost", tr::setUnpreferredCostString);
-        callWith.argument("ignoreRealtimeUpdates", tr::setIgnoreRealtimeUpdates);
-        callWith.argument(
-          "modeWeight",
-          (Map<String, Object> modeWeights) ->
-            tr.setReluctanceForMode(
-              modeWeights
-                .entrySet()
-                .stream()
-                .collect(
-                  Collectors.toMap(e -> TransitMode.valueOf(e.getKey()), e -> (Double) e.getValue())
-                )
-            )
-        );
-      });
-      preferences.withTransfer(tx -> {
-        callWith.argument("transferPenalty", tx::withCost);
-        callWith.argument("minTransferTime", tx::withSlack);
-        callWith.argument("waitReluctance", tx::withWaitReluctance);
-        callWith.argument("maxTransfers", tx::withMaxTransfers);
-        callWith.argument("nonpreferredTransferPenalty", tx::withNonpreferredCost);
-      });
-    });
-
-    callWith.argument("arriveBy", request::setArriveBy);
-
-    callWith.argument(
-      "preferred.routes",
-      request.journey().transit()::setPreferredRoutesFromString
-    );
-
-    callWith.argument(
-      "preferred.agencies",
-      request.journey().transit()::setPreferredAgenciesFromString
-    );
-    callWith.argument(
-      "unpreferred.routes",
-      request.journey().transit()::setUnpreferredRoutesFromString
-    );
-    callWith.argument(
-      "unpreferred.agencies",
-      request.journey().transit()::setUnpreferredAgenciesFromString
-    );
-
-    var transitDisabled = false;
-    if (hasArgument(environment, "banned") || hasArgument(environment, "transportModes")) {
-      var filterRequestBuilder = TransitFilterRequest.of();
-
-      callWith.argument(
-        "banned.routes",
-        s ->
-          filterRequestBuilder.addNot(SelectRequest.of().withRoutesFromString((String) s).build())
-      );
-
-      callWith.argument(
-        "banned.agencies",
-        s ->
-          filterRequestBuilder.addNot(SelectRequest.of().withAgenciesFromString((String) s).build())
+    var args = new GraphQLTypes.GraphQLQueryTypePlanConnectionArgs(environment.getArguments());
+    var dateTime = args.getGraphQLDateTime();
+    if (dateTime.getGraphQLEarliestDeparture() != null) {
```

Comment:
Does the framework validate that you cannot set both latestDeparture and earliestArrival?